syntax = "proto3";

package hubrpc;

// Manager grpc service is used as an entry point for external optimisation
// subsystem to send the equilibrium state.
service Manager {
    //
    // UpdateLink is used to update router link in accordance with givein
    // request. Link might just one channel, or might be the set of
    // channels betwwen user and router. This hook is used by third-parties
    // to put new equilibritum state.
    rpc UpdateLink (UpdateLinkRequest) returns (UpdateLinkResponse);
}

message UpdateLinkRequest {
    // Time of the last sync with state of the router.
    uint64 time = 1;
    uint64 user_id = 2;
    uint64 router_balance = 4;
}

message UpdateLinkResponse {
    // TODO(andrew.shvv) Do we have to return how much it would cost router
    // to apply desired update?
}
